tests/components/watergate/test_config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for the Watergate config flow."""
+
+from homeassistant import data_entry_flow
+from homeassistant.components.watergate.config_flow import WatergateConfigFlow
+from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID
+
+from .const import MOCK_WEBHOOK_ID
+
+from tests.common import HomeAssistant
+
+
+async def test_step_user_form(
+    hass: HomeAssistant, mock_watergate_client, mock_webhook_id_generatio
+) -> None:
+    """Test checking if registration form works end to end."""
+    flow = WatergateConfigFlow()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
```

Comment:
```suggestion
       result = await hass.config_entries.flow.async_init(
            DOMAIN, context={"source": SOURCE_USER}
        )
```
is how this is normally done